biogeochem/EDMortalityFunctionsMod.F90

```diff
@@ -75,12 +81,41 @@ subroutine mortality_rates( cohort_in,bc_in,cmort,hmort,bmort,frmort )
     logical, parameter :: test_zero_mortality = .false. ! Developer test which
                                                         ! may help to debug carbon imbalances
                                                         ! and the like
+     
+   ! Size Dependent Senescence
+    ! rate (r) and inflection point (ip) define the increase in mortality rate with dbh
+    mort_r_size_senescence = EDPftvarcon_inst%mort_r_size_senescence(cohort_in%pft)
+    mort_ip_size_senescence = EDPftvarcon_inst%mort_ip_size_senescence(cohort_in%pft)
+
+    ! if param values have been set then calculate smort
+    if (mort_ip_size_senescence < 10000.0_r8 ) then 
+       smort = 1.0_r8 / ( 1.0_r8 + exp( -1.0_r8 * mort_r_size_senescence * &
+            (cohort_in%dbh - mort_ip_size_senescence) ) ) 
+    else
+       smort = 0.0_r8
+    end if
+
+    ! if user has set cohort age fusion param to not be _ we calculate age
+    ! dependent mortality
+    if (mort_ip_age_senescence < 10000.0_r8) then
```

Comment:
Is it true that we only want to calculate an aging mortality rate if we are tracking aging and cohort_age_tracking is true?  Should we be using that logical here?

Reply:
Yes, ageing mortality can only be calculated if cohort age tracking is on. I changed the logic so that cohort age tracking could be on without having ageing mortality. So now the 'cohort_age_tracking' parameter in the param file turns on cohort age tracking. And both size and age dependent mortality get turned on by setting the ip_age/size_senescence parameters. I can put in some flags so that the ip_age_senescence parameter can't be turned on without cohort_age_tracking being on.